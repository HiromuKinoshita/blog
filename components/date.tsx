import React, { ReactElement } from 'react';
import { parseISO, format } from 'date-fns'
import { dateString } from './date.d'

export default function Date({ dateString }: dateString): ReactElement {
  const date: Date = parseISO(dateString)
  return <time dateTime={dateString}>{format(date, 'LLLL d, yyyy')}</time>
}

if (import.meta.vitest) {
  const { it, expect } = import.meta.vitest
  const d = '20220501'

  it('Date', () => {
    expect(Date({ dateString: d }).props.children).toBe('May 1, 2022')
  })
}
