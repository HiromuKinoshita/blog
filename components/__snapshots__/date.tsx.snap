// Vitest Snapshot v1

exports[`Date 1`] = `
<time
  dateTime="20220501"
>
  May 1, 2022
</time>
`;
